cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -794,36 +814,36 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
         return apiClient.rebalance(host, CruiseControl.REST_API_PORT, rebalanceOptionsBuilder.build(), userTaskID)
                 .map(response -> {
                     if (dryrun) {
-                        if (response.thereIsNotEnoughDataForProposal()) {
-                            // If there are not enough data for a rebalance, it's an actual error at Cruise Control level
-                            // and we need to re-request the proposal at a later stage so we move to the PendingProposal State.
+                        if (!response.isSufficientDataForProposal()) {
+                            // If there is not enough data for a rebalance, it's an error at Cruise Control level
+                            // Need to re-request the proposal at a later time so move to the PendingProposal State.
                             return buildRebalanceStatus(null, KafkaRebalanceState.PendingProposal);
-                        } else if (response.proposalIsStillCalculating()) {
-                            // If rebalance proposal is still being processed we need to re-request the proposal at a later stage
+                        } else if (response.isProposalInProgress()) {
```

Comment:
Most of the time a user will not be asking for some exotic rebalance with strange options and will just get the cached proposal straight back or close to it. The "proposalStillCalaculating" state is assumed to be exceptional so that is why it a flag that has to be explicitly set to true. I am not sure what the advantage is in switching this around.